cmd/add.go

```diff
@@ -294,6 +348,91 @@ func fileDirIsInDirs(file string, dirSet map[string]bool) bool {
 	return false
 }
 
+// handleMeta takes the user provided meta and the backup meta inputs and
+// returns a map containing all valid metadata.
+func handleMeta(meta string, reason Reason, review, removal string) map[string]string {
+	userMeta := parseMetaString(meta)
+
+	mm, err := createBackupMetadata(reason, review, removal, userMeta)
+	if err != nil {
+		die(err.Error()) //nolint:govet
+	}
+
+	return mm
+}
+
+// createBackupMetadata returns a map containing the backup metadata values if
+// the provided inputs are valid.
+func createBackupMetadata(reason Reason, review, removal string, mm map[string]string) (map[string]string, error) {
+	review, removal = setReviewAndRemovalDurations(reason, review, removal)
+
+	removalDate, err := getFutureDateFromDuration(removal)
+	if err != nil {
+		return nil, err
+	}
+
+	reviewDate, err := getFutureDateFromDuration(review)
+	if err != nil {
+		return nil, err
+	}
+
+	if reviewDate.After(removalDate) {
+		die("--review duration must be smaller than --removal duration")
+	}
+
+	mm[put.MetaKeyReason] = Reasons[reason]
+	mm[put.MetaKeyReview] = reviewDate.Format("2006-01-02")
+	mm[put.MetaKeyRemoval] = removalDate.Format("2006-01-02")
+
+	return mm, nil
+}
+
+// setReviewAndRemovalDurations returns the review and removal durations for a
+// given reason. If the user has not set a duration, the function will return
+// the default corresponding to the provided reason.
+func setReviewAndRemovalDurations(reason Reason, review, removal string) (string, string) {
+	defaultReviewDuration, defaultRemovalDuration := getDefaultReviewAndRemovalDurations(reason)
+	if review == "" {
+		review = defaultReviewDuration
+	}
+
+	if removal == "" {
+		removal = defaultRemovalDuration
+	}
+
+	return review, removal
+}
+
+func getDefaultReviewAndRemovalDurations(reason Reason) (string, string) {
+	switch reason.String() {
```

Comment:
No need for the `String()` call here, you can switch on the number and use the `Backup`, `Archive`, and `Quarantine` consts from above in the cases.